File: src/clases/Circulo2D.java
package clases;

/**
 *
 * @author Joel Barrera Quispe
 * 8 de de septiembre de 2025
 * version 1.0
 * Clase Circulo2D 
 */
public class Circulo2D {
    /** atributos centro(x,y) y radio  */
    private double x, y, radio;
    
    /** constructor por default */
    public Circulo2D(){
        this.x = 0;
        this.y = 0;
        this.radio = 1;
    }
    /** constructor con parametros especificados */
    public Circulo2D(double x, double y, double radio){
        this.x = x;
        this.y = y;
        this.radio = radio;
    }

    /**
     * @return the x
     */
    public double getX() {
        return x;
    }

    /**
     * @return the y
     */
    public double getY() {
        return y;
    }

    /**
     * @return the radio
     */
    public double getRadio() {
        return radio;
    }
    
    
    /** calcula el area del circulo */
    public double getArea(){
        return Math.PI * Math.pow(this.getRadio(), 2);
    }
    
    /** calcula el perimetro del circulo */
    public double getPerimetro(){
        return 2* Math.PI * this.radio;
    }
    
    
    /** calcula si el punto dado esta dentro del conjunto de puntos de la circunferencia */
    public boolean contiene(double x, double y){
        double ecuacion = Math.pow( ( x - this.getX() ), 2 ) + Math.pow( ( y - this.getY() ), 2 );
        boolean contiene = (ecuacion < Math.pow(this.radio, 2)) ? true : false;
        return contiene;
    }
    
    
    /** calcula la diferencia entre centros de los circulos, los cuales no deben sobrepasar la diferencia enntre los radios de los dos cirulos */
    public boolean contiene(Circulo2D circulo){
        double calculo = Math.pow( ( (this.getX() - circulo.getX() ) )  ,  2 ) + Math.pow( ( this.getY() - circulo.getY() ) , 2);
        
        boolean contener = ( calculo <= ( this.getRadio() - circulo.getRadio() ) )?true:false;
        
        return contener;
    }
    
    /** calcula la distancia entre puntos y devuelve true si es menor a la suma de los radios, lo contrario devuelve false */
    public boolean sobrepone(Circulo2D circulo){
        double contiene = Math.sqrt( Math.pow( ( this.getX() - circulo.getX() ) , 2)  + Math.pow( ( this.getY() - circulo.getY() ) , x) );
        boolean sobrepone = (contiene < (this.getRadio() + circulo.getRadio())) ? true: false;
        return sobrepone;        
    }
}

File: python/Circulo2D.py
"""
•   PYTHON
 * @author Joel Barrera Quispe
 * 8 de de septiembre de 2025
 * version 1.0
 * Clase Circulo2D 
"""
import math
class Circulo2D:
    # atributos centro(x,y) y radio  */
    def __init__(self, x=0, y=0, radio=1):
        self.centro_x = x
        self.centro_y = y
        self.radio = radio

    # @return the x
    def get_x(self):
        return self.centro_x

    # @return the y
    def get_y(self):
        return self.centro_y

    # @return the radio
    def get_radio(self):
        return self.radio

    #calcula el area del circulo
    def get_area(self):
        return math.pi * (self.get_radio() ** 2)

    #/** calcula el perimetro del circulo */
    def get_perimetro(self):
        return 2 * math.pi * self.radio

    #/** calcula si el punto dado esta dentro del conjunto de puntos de la circunferencia */
    def contiene(self, x, y):
        ecuacion = (x - self.get_x()) ** 2 + (y - self.get_y()) ** 2
        contiene = ecuacion < (self.radio ** 2)
        return contiene

    #/** calcula la diferencia entre centros de los circulos, los cuales no deben sobrepasar la diferencia entre los radios de los dos cirulos */
    def contiene_circulo(self, circulo):
        calculo = (self.get_x() - circulo.get_x()) ** 2 + (self.get_y() - circulo.get_y()) ** 2
        contener = calculo <= (self.get_radio() - circulo.get_radio())
        return contener

    #/** calcula la distancia entre puntos y devuelve true si es menor a la suma de los radios, lo contrario devuelve false */
    def sobrepone(self, circulo):
        contiene = ((self.get_x() - circulo.get_x()) ** 2 + (self.get_y() - circulo.get_y()) ** 2) ** 0.5
        sobrepone = contiene < (self.get_radio() + circulo.get_radio())
        return sobrepone

#/** instanciar el objeto c1, cuyos atributos principales son: centro(2,0) y radio = 1*/
c1 = Circulo2D(2, 0, 1)
#/** calcular el area del mismo circulo */
print("area:", c1.get_area())
#/** calcular el perimetro del circulo */
print("perimetro:", c1.get_perimetro())

#/** operacion contiene: verificar si P(2.5, 0) esta dentro del circulo instanciado */
print("contiene: P(2.5, 0)?", c1.contiene(2.5, 0))
#/** operacion contiene: verificar si el nuevo circulo con atributos: centro(2,0) y radio=0.5, esta dentro dentro del circulo c1*/
print("Contiene: Circulo2D(2, 0, 0.5)?", c1.contiene_circulo(Circulo2D(2, 0, 0.5)))
#/** operacion sobrepone: verificar si el nuevo circulo creado con atributos: c(0,0) y r=2, sobrepone al circulo c1 */
print("Sobrepone: Circulo2D(0, 0, 2)?", c1.sobrepone(Circulo2D(0, 0, 2)))